src/coreclr/System.Private.CoreLib/src/System/Reflection/RtFieldInfo.cs

```diff
@@ -74,7 +74,7 @@ obj is RtFieldInfo fi &&
                 ReferenceEquals(fi.m_reflectedTypeCache.GetRuntimeType(), m_reflectedTypeCache.GetRuntimeType()));
 
         public override int GetHashCode() =>
-            HashCode.Combine(m_fieldHandle.GetHashCode(), m_declaringType.GetUnderlyingNativeHandle().GetHashCode());
+            HashCode.Combine(m_fieldHandle.GetHashCode(), m_declaringType.GetUnderlyingNativeHandle().GetHashCode(), m_reflectedTypeCache.GetRuntimeType().GetHashCode());
```

Comment:
```suggestion
            HashCode.Combine(m_fieldHandle.GetHashCode(), m_declaringType.GetUnderlyingNativeHandle().GetHashCode(), m_reflectedTypeCache.GetRuntimeType().GetUnderlyingNativeHandle().GetHashCode());
```
I think we want to match what we do with m_declaringType for better perf.

Reply:
(Same in the other places.)

Reply:
Done, thanks for the suggestion.